### PR TITLE
Added docker-compose file with master, slave and scheduler.

### PR DESCRIPTION
**Pull request by frankscholten**

### CODE REVIEW

**Comment by mwl** [comment]
Tried `docker-compose up` on mesos/elasticsearch@9796d406e47c5c6852113020abaf4efed03ec2a8

```
Creating elasticsearch_scheduler_1...
Attaching to elasticsearch_master_1, elasticsearch_slave1_1, elasticsearch_scheduler_1
slave1_1    | I0522 09:51:10.377542     1 logging.cpp:172] INFO level logging started!
slave1_1    | I0522 09:51:10.378047     1 main.cpp:142] Build: 2014-11-22 05:29:57 by root
slave1_1    | I0522 09:51:10.378079     1 main.cpp:144] Version: 0.21.0
slave1_1    | I0522 09:51:10.378103     1 main.cpp:147] Git tag: 0.21.0
slave1_1    | I0522 09:51:10.378126     1 main.cpp:151] Git SHA: ab8fa655d34e8e15a4290422df38a18db1c09b5b
slave1_1    | I0522 09:51:10.506139     1 containerizer.cpp:100] Using isolation: posix/cpu,posix/mem
slave1_1    | I0522 09:51:10.511623     1 main.cpp:165] Starting Mesos slave
slave1_1    | I0522 09:51:10.517606    12 slave.cpp:169] Slave started on 1)@127.0.0.1:5051
slave1_1    | W0522 09:51:10.517740    12 slave.cpp:172]
slave1_1    | **************************************************
slave1_1    | Slave bound to loopback interface! Cannot communicate with remote master(s). You might want to set '--ip' flag to a routable IP address.
slave1_1    | **************************************************
slave1_1    | I0522 09:51:10.518429    12 slave.cpp:289] Slave resources: ports(*):[9200-9200, 9300-9300]; cpus(*):4; mem(*):498; disk(*):13483
slave1_1    | I0522 09:51:10.518564    12 slave.cpp:318] Slave hostname: es-mesos
slave1_1    | I0522 09:51:10.518654    12 slave.cpp:319] Slave checkpoint: true
slave1_1    | I0522 09:51:10.524723    10 state.cpp:33] Recovering state from '/tmp/mesos/meta'
slave1_1    | I0522 09:51:10.525490     8 status_update_manager.cpp:197] Recovering status update manager
slave1_1    | I0522 09:51:10.526191    10 docker.cpp:767] Recovering Docker containers
slave1_1    | I0522 09:51:10.549964     6 containerizer.cpp:281] Recovering containerizer
slave1_1    | I0522 09:51:10.709333     9 slave.cpp:3466] Finished recovery
slave1_1    | I0522 09:51:10.712383     9 slave.cpp:602] New master detected at master@127.0.0.1:5050
slave1_1    | I0522 09:51:10.712621     6 status_update_manager.cpp:171] Pausing sending status updates
slave1_1    | I0522 09:51:10.713405     9 slave.cpp:627] No credentials provided. Attempting to register without authentication
slave1_1    | I0522 09:51:10.713537     9 slave.cpp:638] Detecting new master
master_1    | I0522 09:50:50.021941     1 logging.cpp:172] INFO level logging started!
master_1    | I0522 09:50:50.023591     1 main.cpp:167] Build: 2014-11-22 05:29:57 by root
master_1    | I0522 09:50:50.023613     1 main.cpp:169] Version: 0.21.0
master_1    | I0522 09:50:50.023627     1 main.cpp:172] Git tag: 0.21.0
master_1    | I0522 09:50:50.023639     1 main.cpp:176] Git SHA: ab8fa655d34e8e15a4290422df38a18db1c09b5b
master_1    | I0522 09:50:50.032721     1 leveldb.cpp:176] Opened db in 8.557311ms
master_1    | I0522 09:50:50.033978     1 leveldb.cpp:183] Compacted db in 1.131812ms
master_1    | I0522 09:50:50.034096     1 leveldb.cpp:198] Created db iterator in 54551ns
master_1    | I0522 09:50:50.034129     1 leveldb.cpp:204] Seeked to beginning of db in 5241ns
master_1    | I0522 09:50:50.034148     1 leveldb.cpp:273] Iterated through 0 keys in the db in 2060ns
master_1    | I0522 09:50:50.034379     1 replica.cpp:741] Replica recovered with log positions 0 -> 0 with 1 holes and 0 unlearned
master_1    | I0522 09:50:50.036950     9 recover.cpp:437] Starting replica recovery
master_1    | I0522 09:50:50.037663     1 main.cpp:292] Starting Mesos master
master_1    | I0522 09:50:50.037755     9 recover.cpp:463] Replica is in EMPTY status
master_1    | I0522 09:50:50.040576     6 replica.cpp:638] Replica in EMPTY status received a broadcasted recover request
master_1    | I0522 09:50:50.041921    10 recover.cpp:188] Received a recover response from a replica in EMPTY status
master_1    | I0522 09:50:50.043313     8 recover.cpp:554] Updating replica status to STARTING
master_1    | I0522 09:50:50.043520     1 master.cpp:318] Master 20150522-095050-16777343-5050-1 (es-mesos) started on 127.0.0.1:5050
master_1    | W0522 09:50:50.044208     1 master.cpp:322]
master_1    | **************************************************
master_1    | Master bound to loopback interface! Cannot communicate with remote schedulers or slaves. You might want to set '--ip' flag to a routable IP address.
master_1    | **************************************************
master_1    | I0522 09:50:50.045677     1 master.cpp:366] Master allowing unauthenticated frameworks to register
master_1    | I0522 09:50:50.046665     1 master.cpp:371] Master allowing unauthenticated slaves to register
master_1    | I0522 09:50:50.047966    10 leveldb.cpp:306] Persisting metadata (8 bytes) to leveldb took 4.21253ms
master_1    | I0522 09:50:50.048044    10 replica.cpp:320] Persisted replica status to STARTING
master_1    | I0522 09:50:50.048454     6 recover.cpp:463] Replica is in STARTING status
master_1    | I0522 09:50:50.049872     9 replica.cpp:638] Replica in STARTING status received a broadcasted recover request
master_1    | I0522 09:50:50.050307     8 recover.cpp:188] Received a recover response from a replica in STARTING status
master_1    | I0522 09:50:50.050784     6 recover.cpp:554] Updating replica status to VOTING
master_1    | I0522 09:50:50.052630     6 leveldb.cpp:306] Persisting metadata (8 bytes) to leveldb took 1.379311ms
master_1    | I0522 09:50:50.052752     6 replica.cpp:320] Persisted replica status to VOTING
master_1    | I0522 09:50:50.053094     8 recover.cpp:568] Successfully joined the Paxos group
master_1    | I0522 09:50:50.053342     8 recover.cpp:452] Recover process terminated
master_1    | I0522 09:50:50.055238     1 master.cpp:1202] Successfully attached file '/var/log/mesos-master.INFO'
master_1    | I0522 09:50:50.055562    11 master.cpp:1263] The newly elected leader is master@127.0.0.1:5050 with id 20150522-095050-16777343-5050-1
master_1    | I0522 09:50:50.055620    11 master.cpp:1276] Elected as the leading master!
master_1    | I0522 09:50:50.055667    11 master.cpp:1094] Recovering from registrar
master_1    | I0522 09:50:50.056025    12 registrar.cpp:313] Recovering registrar
master_1    | I0522 09:50:50.057430    11 log.cpp:656] Attempting to start the writer
master_1    | I0522 09:50:50.061383     5 replica.cpp:474] Replica received implicit promise request with proposal 1
master_1    | I0522 09:50:50.062002     5 leveldb.cpp:306] Persisting metadata (8 bytes) to leveldb took 515771ns
master_1    | I0522 09:50:50.062072     5 replica.cpp:342] Persisted promised to 1
master_1    | I0522 09:50:50.063544     5 coordinator.cpp:230] Coordinator attemping to fill missing position
master_1    | I0522 09:50:50.066402     5 replica.cpp:375] Replica received explicit promise request for position 0 with proposal 2
master_1    | I0522 09:50:50.067069     5 leveldb.cpp:343] Persisting action (8 bytes) to leveldb took 568011ns
master_1    | I0522 09:50:50.067261     5 replica.cpp:676] Persisted action at 0
master_1    | I0522 09:50:50.069452     7 replica.cpp:508] Replica received write request for position 0
master_1    | I0522 09:50:50.069602     7 leveldb.cpp:438] Reading position from leveldb took 66059ns
master_1    | I0522 09:50:50.070188     7 leveldb.cpp:343] Persisting action (14 bytes) to leveldb took 407239ns
master_1    | I0522 09:50:50.070276     7 replica.cpp:676] Persisted action at 0
master_1    | I0522 09:50:50.071130     7 replica.cpp:655] Replica received learned notice for position 0
master_1    | I0522 09:50:50.071663     7 leveldb.cpp:343] Persisting action (16 bytes) to leveldb took 448909ns
master_1    | I0522 09:50:50.071719     7 replica.cpp:676] Persisted action at 0
master_1    | I0522 09:50:50.071751     7 replica.cpp:661] Replica learned NOP action at position 0
master_1    | I0522 09:50:50.072548     6 log.cpp:672] Writer started with ending position 0
master_1    | I0522 09:50:50.075350     7 leveldb.cpp:438] Reading position from leveldb took 61945ns
master_1    | I0522 09:50:50.079190    10 registrar.cpp:346] Successfully fetched the registry (0B) in 23.039232ms
master_1    | I0522 09:50:50.079567    10 registrar.cpp:445] Applied 1 operations in 27886ns; attempting to update the 'registry'
master_1    | I0522 09:50:50.086057     9 log.cpp:680] Attempting to append 114 bytes to the log
master_1    | I0522 09:50:50.089800     8 coordinator.cpp:340] Coordinator attempting to write APPEND action at position 1
master_1    | I0522 09:50:50.091292     7 replica.cpp:508] Replica received write request for position 1
master_1    | I0522 09:50:50.092039     7 leveldb.cpp:343] Persisting action (130 bytes) to leveldb took 505332ns
master_1    | I0522 09:50:50.092118     7 replica.cpp:676] Persisted action at 1
master_1    | I0522 09:50:50.093032     5 replica.cpp:655] Replica received learned notice for position 1
master_1    | I0522 09:50:50.093708     5 leveldb.cpp:343] Persisting action (133 bytes) to leveldb took 548165ns
master_1    | I0522 09:50:50.093765     5 replica.cpp:676] Persisted action at 1
master_1    | I0522 09:50:50.093802     5 replica.cpp:661] Replica learned APPEND action at position 1
master_1    | I0522 09:50:50.095384     6 registrar.cpp:490] Successfully updated the 'registry' in 15.635968ms
master_1    | I0522 09:50:50.095852     6 registrar.cpp:376] Successfully recovered registrar
master_1    | I0522 09:50:50.095861    10 log.cpp:699] Attempting to truncate the log to 1
master_1    | I0522 09:50:50.096220     5 coordinator.cpp:340] Coordinator attempting to write TRUNCATE action at position 2
master_1    | I0522 09:50:50.096284    10 master.cpp:1121] Recovered 0 slaves from the Registry (78B) ; allowing 10mins for slaves to re-register
master_1    | I0522 09:50:50.097401     7 replica.cpp:508] Replica received write request for position 2
master_1    | I0522 09:50:50.097954     7 leveldb.cpp:343] Persisting action (16 bytes) to leveldb took 482223ns
master_1    | I0522 09:50:50.097991     7 replica.cpp:676] Persisted action at 2
master_1    | I0522 09:50:50.098498    12 replica.cpp:655] Replica received learned notice for position 2
master_1    | I0522 09:50:50.099026    12 leveldb.cpp:343] Persisting action (18 bytes) to leveldb took 454092ns
master_1    | I0522 09:50:50.099092    12 leveldb.cpp:401] Deleting ~1 keys from leveldb took 28237ns
master_1    | I0522 09:50:50.099114    12 replica.cpp:676] Persisted action at 2
master_1    | I0522 09:50:50.099148    12 replica.cpp:661] Replica learned TRUNCATE action at position 2
scheduler_1 | Exception in thread "main" java.lang.UnsupportedClassVersionError: org/apache/mesos/elasticsearch/scheduler/ElasticsearchScheduler : Unsupported major.minor version 52.0
scheduler_1 | 	at java.lang.ClassLoader.defineClass1(Native Method)
scheduler_1 | 	at java.lang.ClassLoader.defineClass(ClassLoader.java:800)
scheduler_1 | 	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
scheduler_1 | 	at java.net.URLClassLoader.defineClass(URLClassLoader.java:449)
scheduler_1 | 	at java.net.URLClassLoader.access$100(URLClassLoader.java:71)
scheduler_1 | 	at java.net.URLClassLoader$1.run(URLClassLoader.java:361)
scheduler_1 | 	at java.net.URLClassLoader$1.run(URLClassLoader.java:355)
scheduler_1 | 	at java.security.AccessController.doPrivileged(Native Method)
scheduler_1 | 	at java.net.URLClassLoader.findClass(URLClassLoader.java:354)
scheduler_1 | 	at java.lang.ClassLoader.loadClass(ClassLoader.java:425)
scheduler_1 | 	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:308)
scheduler_1 | 	at java.lang.ClassLoader.loadClass(ClassLoader.java:358)
scheduler_1 | 	at sun.launcher.LauncherHelper.checkAndLoadMain(LauncherHelper.java:482)
elasticsearch_scheduler_1 exited with code 1
Gracefully stopping... (press Ctrl+C again to force)
Stopping elasticsearch_slave1_1...
Stopping elasticsearch_master_1...
```

**Comment by frankscholten** [comment]
Made some changes to docker-compose and changed the scheduler so it maps the hostname of the master to an IP address so it can be used by the elasticsearch container.

However when I start everything and run docker logs it complains it cannot find <b>cluster</b> in the state file. Is this a regression?

```
[2015-05-22 15:07:43,126][ERROR][discovery.mesos          ] [Ent] unexpected error while joining cluster, trying again
org.elasticsearch.ElasticsearchException: Ping execution failed
	at org.elasticsearch.discovery.zen.ping.unicast.UnicastZenPing.ping(UnicastZenPing.java:263)
	at org.elasticsearch.discovery.zen.ping.ZenPingService.ping(ZenPingService.java:146)
	at org.elasticsearch.discovery.zen.ping.ZenPingService.pingAndWait(ZenPingService.java:124)
	at org.elasticsearch.discovery.zen.ZenDiscovery.findMaster(ZenDiscovery.java:943)
	at org.elasticsearch.discovery.zen.ZenDiscovery.innerJoinCluster(ZenDiscovery.java:337)
	at org.elasticsearch.discovery.zen.ZenDiscovery.access$6000(ZenDiscovery.java:80)
	at org.elasticsearch.discovery.zen.ZenDiscovery$JoinThreadControl$1.run(ZenDiscovery.java:1320)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: org.json.JSONException: JSONObject["cluster"] not found.
	at org.json.JSONObject.get(JSONObject.java:473)
	at org.json.JSONObject.getString(JSONObject.java:654)
	at org.elasticsearch.cloud.mesos.MesosStateServiceRest.getNodeIpsAndPorts(MesosStateServiceRest.java:54)
	at org.elasticsearch.discovery.mesos.MesosUnicastHostsProvider.buildDynamicNodes(MesosUnicastHostsProvider.java:37)
	at org.elasticsearch.discovery.zen.ping.unicast.UnicastZenPing.sendPings(UnicastZenPing.java:316)
	at org.elasticsearch.discovery.zen.ping.unicast.UnicastZenPing.ping(UnicastZenPing.java:224)
	... 9 more
```

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  hostname: slave1
+  links:
+   - mesosmaster
+  environment:
+    - MESOS_MASTER=mesosmaster:5050
+    - MESOS_EXECUTOR_REGISTRATION_TIMEOUT=5mins
+    - MESOS_CONTAINERIZERS=docker,mesos
+    - MESOS_ISOLATOR=cgroups/cpu,cgroups/mem
+    - MESOS_LOG_DIR=/var/log
+    - MESOS_RESOURCES=ports(*):[9200-9200,9300-9300]
```
Comment (lguminski) [inline docker-compose.yml]: Is this a way to express that only 2 ports will be used? 9200 and 9300 ?

Comment (frankscholten) [inline docker-compose.yml]: Yes. I am currently changing the scheduler so it takes the first two ports from a series of ranges and uses them as the client and transport port. When that is done we can add multiple slaves in the docker-compose file.

**Pull request merged**